core/src/property/backend/lww.rs

```diff
@@ -62,36 +56,34 @@ impl PropertyBackend for LWWBackend {
         Box::new(Self { values: Arc::new(RwLock::new(cloned)) })
     }
 
-    fn properties(&self) -> Vec<String> {
+    fn properties(&self) -> Vec<PropertyName> {
         let values = self.values.read().unwrap();
-        values.keys().cloned().collect::<Vec<String>>()
+        values.keys().cloned().collect::<Vec<PropertyName>>()
     }
 
-    fn materialized(&self) -> BTreeMap<PropertyName, Materialized> { unimplemented!() }
+    fn property_values(&self) -> BTreeMap<PropertyName, PropertyValue> {
+        let values = self.values.read().unwrap();
+        values.clone()
+    }
 
     fn property_backend_name() -> String { "lww".to_owned() }
 
+    // This is identical to [`to_operations`] for [`LWWBackend`].
     fn to_state_buffer(&self) -> anyhow::Result<Vec<u8>> {
-        let values = self.values.read().unwrap();
-        let state_buffer = bincode::serialize(&*values)?;
+        let property_values = self.property_values();
+        let state_buffer = bincode::serialize(&property_values)?;
         Ok(state_buffer)
     }
 
     fn from_state_buffer(state_buffer: &Vec<u8>) -> std::result::Result<Self, crate::error::RetrievalError>
     where Self: Sized {
-        let map = bincode::deserialize::<BTreeMap<PropertyName, Vec<u8>>>(state_buffer)?;
+        let map = bincode::deserialize::<BTreeMap<PropertyName, PropertyValue>>(state_buffer)?;
         Ok(Self { values: Arc::new(RwLock::new(map)) })
     }
 
     fn to_operations(&self) -> anyhow::Result<Vec<super::Operation>> {
-        let values = self.values.read().unwrap();
-
-        let mut map = BTreeMap::<&PropertyName, &Vec<u8>>::default();
-        for (property_name, value) in &*values {
-            map.insert(property_name, value);
-        }
-
-        let serialized_diff = bincode::serialize(&map)?;
+        let property_values = self.property_values();
+        let serialized_diff = bincode::serialize(&property_values)?;
```

Comment:
We really want to be storing deltas here of some kind.

For now I suggest we at least find a way to version this so we don't have to do migrations when we later change this to a delta format

Bincode kiiind of isn't the right thing for this, but we can probably get away with it if we wrap it in a Version enum

eg
```
enum Version{
   V1(Whatever)
   // V2 add this later
 }
```

It's not great, but it should work as a band-aid
https://users.rust-lang.org/t/is-bincode-compatible-with-appending-enum-variants/40084

Reply:
I created a
```rust
struct LWWDiff {
	version: u8,
	data: Vec<u8>,
}
```
instead of the enum just because I don't think serde/bincode guarantees anything with the enum additions. But should work the same way.